core/src/main/java/jenkins/console/ConsoleUrlProviderGlobalConfiguration.java

```diff
@@ -0,0 +1,92 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.console;
+
+import hudson.BulkChange;
+import hudson.Extension;
+import hudson.ExtensionList;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Allows administrators to activate and sort {@link ConsoleUrlProvider} extensions to set defaults for all users.
+ * @see ConsoleUrlProviderUserProperty
+ * @since TODO
+ */
+@Extension
+@Symbol("consoleUrlProvider")
+@Restricted(NoExternalUse.class)
+public class ConsoleUrlProviderGlobalConfiguration extends GlobalConfiguration {
+    private static final Logger LOGGER = Logger.getLogger(ConsoleUrlProviderGlobalConfiguration.class.getName());
+
+    private List<ConsoleUrlProvider> providers;
+
+    public ConsoleUrlProviderGlobalConfiguration() {
+        load();
+    }
+
+    public List<ConsoleUrlProvider> getProviders() {
+        return providers;
+    }
+
+    @DataBoundSetter
+    public void setProviders(List<ConsoleUrlProvider> providers) {
+        this.providers = providers;
+        save();
+    }
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        // We have to null out providers before data binding to allow all providers to be deleted in the config UI.
+        // We use a BulkChange to avoid double saves in other cases.
+        try (BulkChange bc = new BulkChange(this)) {
+            providers = null;
+            req.bindJSON(this, json);
+            bc.commit();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Failed to save " + getConfigFile(), e);
+        }
+        return true;
+    }
+
+    public boolean isEnabled() {
+        return !Jenkins.get().getDescriptorList(ConsoleUrlProvider.class).isEmpty();
```

Comment:
won't there always be at least one? shouldn't it be if the only 1 isn't the default one?

Reply:
Yes good catch, this was before I added the default extension. I'll update it.

Reply:
Fixed in [5f051c9](https://github.com/jenkinsci/jenkins/pull/8321/commits/5f051c9b0faf10e39f48122389f38b05ce2b3ef6).